src/accounts/accounts.controller.ts

```diff
@@ -122,4 +125,36 @@ export class AccountsController {
     const permissions = await this.accountsService.getPermissions(account);
     return createPermissionsModel(permissions);
   }
+
+  @ApiOperation({
+    summary: 'Get Account Subsidy',
+    description:
+      'The endpoint retrieves the subsidized balance of this Account and the subsidizer Account',
+  })
+  @ApiParam({
+    name: 'account',
+    description: 'The Account address whose subsidy is to be fetched',
+    type: 'string',
+    example: '5GwwYnwCYcJ1Rkop35y7SDHAzbxrCkNUDD4YuCUJRPPXbvyV',
+  })
+  @ApiOkResponse({
+    description: 'Subsidy details for the Account',
+    type: SubsidyModel,
+  })
+  @ApiNoContentResponse({
+    description: 'Account is not being subsidized',
+  })
+  @ApiNotFoundResponse({
+    description: 'Account is not associated with any Identity',
+  })
```

Comment:
I think this *should* throw BadRequest for bad SS58 format, but its returning a InternalServerError in that case.

I think that error handling should be moved into the shared function instead of accounts service

Reply:
This was already handled in `findOne` method. I tested it out. Apparently await was missing which was causing this